java-frontend/src/main/java/org/sonar/java/se/xproc/BehaviorCache.java

```diff
@@ -72,6 +74,23 @@ public MethodBehavior get(Symbol.MethodSymbol symbol) {
     return behaviors.get(symbol);
   }
 
+  private static boolean isRequireNonNullMethod(Symbol.MethodSymbol symbol) {
+    return isObjectsRequireNonNullMethod(symbol) || isValidateMethod(symbol) || isLog4jOrSpringAssertNotNull(symbol);
+  }
+
+  private static boolean isValidateMethod(Symbol.MethodSymbol symbol) {
+    Type type = symbol.owner().type();
+    String name = symbol.name();
+    return (type.is("org.apache.commons.lang3.Validate") || type.is("org.apache.commons.lang.Validate"))
+      && ("notEmpty".equals(name) || "notNull".equals(name));
+  }
+
+  private static boolean isCollectionUtilsIsEmpty(Symbol.MethodSymbol symbol) {
+    Type type = symbol.owner().type();
+    return (type.is("org.apache.commons.collections4.CollectionUtils") || type.is("org.apache.commons.collections.CollectionUtils"))
+      && "isEmpty".equals(symbol.name());
```

Comment:
I would also include [CollectionUtils.isNotEmpty()](http://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/CollectionUtils.html#isNotEmpty(java.util.Collection)), as it's also a null-safe check, with following method yields in case of `false` result :

- result is `false`, arg0 is `null`;
- result is `false`, arg0 is `not-null` ( and `empty` )
- result is `true`, arg0 is `not-null` (and `not-empty`)